src/config.c

```diff
@@ -1693,14 +1693,16 @@ config_check ( void )
 
 static int config_newcfg = 0;
 
-static char *config_get_dir ( void )
```

Comment:
When I was testing this (in a docker container mind you) I didn't have any issue with this. I don't think this is a bad idea of course, I didn't look into this too deeply!

Regardless, this should be its own commit, not connected to the debian packaging :)

Reply:
Were you using the forking mode of operation when you tested?  The problem I encountered is that the Debian package forks (starts as root and then drops privileges), so getuid() still returns 0 when the config directory is checked.  In the case that forking is not used, the passed uid will be -1 and getuid() will then be used as you originally wrote.

Reply:
Ah! No! I did look into `geteuid()` (afaik) but your solutions is cleaner anyway. I was testing the docker container, which will run tvheadend as <user> anyhow.

Thanks for the catch!